### ISSUE TITLE
Clicking inside a block the block setting should automatically be seen.

### ISSUE THREAD

**Issue by paaljoachim** [WordPress/gutenberg#5064]
I am testing version 2.1

Clicking inside a block in the preview area the right hand side block setting should automatically be seen.If the right side Document setting is selected it will also stay selected even when clicking or adding a block. As a block is added or clicked into the block setting should be seen so the user can see what can be done inside the block setting in relation to the block being used.

When the user needs to adjust something in the document setting the user can then click the document setting tab and switch over to it.

Bottom line: Relationships between what is being done and the settings/options in relation to it. Working on the document then document settings can be seen. Working on individual blocks then block settings can be seen. Just click the other tab to switch over.

![screen shot 2018-02-15 at 00 17 23](https://user-images.githubusercontent.com/5323259/36241643-9fb5e232-11e5-11e8-8028-76354160b7c2.png)

**Comment by karmatosed** [comment]
This has been gone through before and was ultimately decided to work way it does. That said, I still think it's worth considering what if the sidebar is closed. In that case I think that having it not show makes sense. If it is open then going straight to block settings does make sense. These were my feelings before and stand regarding this.

**Comment by mtias** [comment]
Yes, this was implemented in https://github.com/WordPress/gutenberg/pull/819 a long time ago, then changed in https://github.com/WordPress/gutenberg/pull/1833 as the general user feedback was it was confusing to have mode switch to the block whenever you selected a block. It could be revisited as the sidebar patterns are more settled.

**Comment by mtias** [comment]
@jasmussen what do you think of revisiting this?

**Comment by jasmussen** [comment]
As you know, I love this idea, so much so that it was a core value of the initial inspector design (and the breadcrumb tree style in the top tab). 

The reason we reverted was because of pushback. 

I don't know if the tabs change anything, with regards to the pushback — if the Block tab was auto-selected when focusing a block and the sidebar was open, presumably clicking the _Document_ tab would _deselect the block_, just like the breadcrumb trail did. 

Or is there a nuance we can explore here? 

CC: @folletto in case he has any ideas.

**Comment by karmatosed** [comment]
I would still wave the +1 flag to bringing this back :) 🚩

> I don't know if the tabs change anything, with regards to the pushback — if the Block tab was auto-selected when focusing a block and the sidebar was open, presumably clicking the Document tab would deselect the block, just like the breadcrumb trail did.

I would +1 to only selecting the blocks tab, this to me is the right nuance. I also think clicking Document should deselect as you are moving out of the flow.

**Comment by folletto** [comment]
> The reason we reverted was because of pushback.

My take would be that 10 Jul 2017  was too early to see the benefit of the blocks sidebar getting focus when a block was selected: I might be entirely mistaken, but at that time a lot of block sidebars were still empty or with the same content of the toolbar.

I think now there's enough richness to show that the having full interaction consistency — when manipulating an object the sidebar should be showing that object inspector — also has direct, solid benefits.

I'd +1 too the nuance of behaviour expressed by @jasmussen above. Same thing, said in a different way:

* Sidebar closed = selecting doesn't do anything on it
* Sidebar opened + No block selected = Document sidebar active
* Sidebar opened + Block selected = Block sidebar active
* Explicitly clicking on Document = De-selects the block

Sounds good to me.

**Comment by mtias** [comment]
This has been added back.
